update-arcdps.ps1

```diff
@@ -42,248 +44,107 @@ if (-not $config.dll_backup_path) {
-    # this file has the md5sum followed by a filename
-    $web_md5 = $web_md5.toString().trim().split(" ")[0].toUpper()
-    Write-Host "arcdps: Online MD5 Hash:  $web_md5"
-
-    if ($current_md5 -ne $web_md5) {
-        $run_update = $true
-    }
-} else {
-    $run_update = $true
-}
-
-if ($run_update -eq $false) {
-    Write-Host "Current version is up to date"
-} else {
-    # If we have a copy of ArcDPS, make a new backup before overwriting
-    if (X-Test-Path $arc_path) {
-        if (X-Test-Path $arc_backup) {
-            Remove-Item $arc_backup
-        }
-        Move-Item $arc_path $arc_backup
-    }
-
-    # Also backup the templates plugin
-    if (X-Test-Path $templates_path) {
-        if (X-Test-Path $templates_backup) {
-            Remove-Item $templates_backup
-        }
-        Move-Item $templates_path $templates_backup
-    }
-
-    # Also backup extras
-    if (X-Test-Path $extras_path) {
-        if (X-Test-Path $extras_backup) {
-            Remove-Item $extras_backup
-        }
-        Move-Item $extras_path $extras_backup
-    }
-
-    # Remove the copy in bin64 as well
-    if (X-Test-Path $arc_bin_path) {
-        Remove-Item $arc_bin_path
-    }
-    if (X-Test-Path $templates_bin_path) {
-        Remove-Item $templates_bin_path
-    }
-    if (X-Test-Path $extras_bin_path) {
-        Remove-Item $extras_bin_path
-    }
-
-    Write-Host "Downloading new arcdps d3d9.dll"
-    Invoke-WebRequest -Uri $arc_url -UseBasicParsing -OutFile $arc_path
-    Copy-Item $arc_path $arc_bin_path
-    Write-Host "Downloading new arcdps d3d9_arcdps_build_templates.dll"
-    Invoke-WebRequest -Uri $templates_url -UseBasicParsing -OutFile $templates_path
-    Copy-Item $templates_path $templates_bin_path
-    Write-Host "Downloading new arcdps d3d9_arcdps_extras.dll"
-    Invoke-WebRequest -Uri $extras_url -UseBasicParsing -OutFile $extras_path
-    Copy-Item $extras_path $extras_bin_path
-}
-
-$run_update = $false
-Write-Host "Checking d3d9_arcdps_mechanics.dll MD5 Hash for changes"
-if (X-Test-Path $mechanics_path) {
-    $current_md5 = (Get-FileHash $mechanics_path -Algorithm MD5).Hash
-    Write-Host "mechanics: Current MD5 Hash: $current_md5"
-    $web_md5 = Invoke-WebRequest -URI $mechanics_md5_url -UseBasicParsing
-    # file is just the md5sum, without a filename
+$files = @(
+    @{
+        name = 'ArcDPS'
+        dll = 'd3d9.dll'
+        backup = 'arc-d3d9.dll.back'
+        url = 'https://www.deltaconnected.com/arcdps/x64/d3d9.dll'
+        md5_url = 'https://www.deltaconnected.com/arcdps/x64/d3d9.dll.md5sum'
+        md5_file = 'd3d9.dll'
+    },
+    @{
+        name = 'Extras Extension'
+        dll = 'd3d9_arcdps_extras.dll'
+        backup = 'extension-d3d9_arcdps_extras.dll.back'
+        url = 'https://www.deltaconnected.com/arcdps/x64/extras/d3d9_arcdps_extras.dll'
+        md5_url = 'https://www.deltaconnected.com/arcdps/x64/d3d9.dll.md5sum'
```

Comment:
every of the other DLLs use this md5sum file?

Reply:
we shouldn't download the md5 over and over.

Reply:
They don't all use the same md5sum. Only the arcdps, extras, and buildtemplates ones do, because they don't have their own md5 URL.

I think I have a better way to fix this though.